core/src/reporter/state.ts

```diff
@@ -294,35 +316,31 @@ export class State {
         throw new OraklError(OraklErrorCode.WalletNotActive, msg)
       }
 
+      // Assumption: the only source of error can be json-rpc call
+      // Solution: keep polling/retrying until json-rpc becomes responsive
+      // If successful, the "break" statement will break the infinite loop
+      let retryCount = 0
       let remoteNonce: number
-      try {
-        if (this.delegatedFee) {
-          const caverWallet = wallet as CaverWallet
-          remoteNonce = Number(
-            await caverWallet.caver.rpc.klay.getTransactionCount(caverWallet.address)
-          )
-        } else {
-          remoteNonce = await (wallet as NonceManager).getTransactionCount()
+      while (true) {
+        try {
+          remoteNonce = await this.getRemoteNonce(wallet)
+          break
+        } catch (error) {
+          // Slack the error message every half an hour
+          if (retryCount % NONCE_MANAGER_SLACK_FREQUENCY_RETRIES === 0) {
+            this.logger.error(
+              { name: 'getAndIncrementNonce', file: FILE_NAME },
+              `Error while fetching nonce for oracle ${oracleAddress}. Retrying...`
+            )
+            await sendToSlack(error)
+          }
+          retryCount += 1
+
+          // Sleep before retrying
+          await sleep(NONCE_MANAGER_POLLING_INTERVAL)
         }
-      } catch (error) {
-        const msg = `Failed to get nonce for wallet`
-        this.logger.error({ name: 'getAndIncrementNonce', file: FILE_NAME }, msg)
-        throw new OraklError(OraklErrorCode.FailedToGetWalletTransactionCount, msg)
       }
-
-      const localNonce = this.nonces[oracleAddress]
-
-      let nonce: number
-      if (!localNonce || remoteNonce > localNonce) {
-        this.logger.warn(
-          { name: 'getAndIncrementNonce', file: FILE_NAME },
-          `Nonce value discrepancy. Remote: ${remoteNonce}, Local: ${localNonce}. Updating local nonce to remote nonce.`
-        )
-        nonce = remoteNonce
-      } else {
-        nonce = localNonce
-      }
-
+      const nonce = Math.max(this.nonces[oracleAddress], remoteNonce)
```

Comment:
@Intizar-T It seems that we fetch remote nonce for every new transaction and if we cannot fetch it we keep trying until the server responds. If that is true, I am not sure the local nonce mapping `this.nonces` has real purpose. Can you please share why do we need it? 🙏

Reply:
good point Martin. I was trying to take care of a case where the wallet that we use to report makes a transaction outside this service (vrf or rr) and my local nonce will be outdated. Would you say that's a realistic scenario? Do we use the reporter wallets anywhere else outside the vrf/rr services? If not then it'll make things much easier. If yes, then I'll need to come up with a different approach to solve it.

Reply:
The only time we use it when we want to take earned KLAY out of the reporter, but that does not happen often. And if that happens, the current solution should fail and get new nonce reassigned.

Reply:
got it. Then I'll simply catch the `NonceAlreadyUsed` error and retry with new nonce